src/net/socket.rs

```diff
@@ -60,34 +59,40 @@ impl Socket {
     /// endpoint by looking to see if they are still sending packets or not
     ///
     /// This function allows you to configure laminar with the passed configuration.
-    pub fn bind_with_config<A: ToSocketAddrs>(
-        addresses: A,
-        config: Config,
-    ) -> Result<(Self, Sender<Packet>, Receiver<SocketEvent>)> {
+    pub fn bind_with_config<A: ToSocketAddrs>(addresses: A, config: Config) -> Result<Self> {
         let socket = UdpSocket::bind(addresses)?;
         Self::bind_internal(socket, config)
     }
 
-    fn bind_internal(
-        socket: UdpSocket,
-        config: Config,
-    ) -> Result<(Self, Sender<Packet>, Receiver<SocketEvent>)> {
+    fn bind_internal(socket: UdpSocket, config: Config) -> Result<Self> {
         socket.set_nonblocking(true)?;
         let (event_sender, event_receiver) = unbounded();
         let (packet_sender, packet_receiver) = unbounded();
-        Ok((
-            Socket {
-                recv_buffer: vec![0; config.receive_buffer_max_size],
-                socket,
-                config,
-                connections: ActiveConnections::new(),
-                link_conditioner: None,
-                event_sender,
-                packet_receiver,
-            },
-            packet_sender,
-            event_receiver,
-        ))
+        Ok(Socket {
+            recv_buffer: vec![0; config.receive_buffer_max_size],
+            socket,
+            config,
+            connections: ActiveConnections::new(),
+            link_conditioner: None,
+            event_sender,
+            packet_receiver,
+
+            sender: packet_sender,
+            receiver: event_receiver,
+        })
+    }
+
+    /// Send a packet
+    pub fn send(&mut self, packet: Packet) {
+        let _ = self.sender.send(packet);
```

Comment:
Probably shouldn't swallow the error here. Would be worth transforming it into a Laminar specific error.